File: LMIS/LMIS.Web/Modules/ReaderManage/CardLevels/CardLevelsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.ReaderManage.Columns;

[ColumnsScript("ReaderManage.CardLevels")]
[BasedOnRow(typeof(CardLevelsRow), CheckNames = true)]
public class CardLevelsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public long CardLevelId { get; set; }
    [EditLink]
    public string CardLevelName { get; set; }
    public decimal Discount { get; set; }
    public decimal Penalty { get; set; }
    public int BorrowDays { get; set; }
    public int BorrowBooks { get; set; }
    public DateTime CreateTime { get; set; }
    public DateTime UpdateTime { get; set; }
}

File: LMIS/LMIS.Web/Modules/ServerTypes/ReaderManage/CardlevelsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CardLevelsRow } from "./CardLevelsRow";

export interface CardLevelsColumns {
    CardLevelId: Column<CardLevelsRow>;
    CardLevelName: Column<CardLevelsRow>;
    Discount: Column<CardLevelsRow>;
    Penalty: Column<CardLevelsRow>;
    BorrowDays: Column<CardLevelsRow>;
    BorrowBooks: Column<CardLevelsRow>;
    CreateTime: Column<CardLevelsRow>;
    UpdateTime: Column<CardLevelsRow>;
}

export class CardLevelsColumns extends ColumnsBase<CardLevelsRow> {
    static readonly columnsKey = 'ReaderManage.CardLevels';
    static readonly Fields = fieldsProxy<CardLevelsColumns>();
}